pkg/controller/crdmirroring/crdhandler/clustergroup.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crdhandler
+
+import (
+	"context"
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	crd "github.com/vmware-tanzu/antrea/pkg/apis/crd/v1alpha2"
+	crdclientset "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	crdlister "github.com/vmware-tanzu/antrea/pkg/client/listers/crd/v1alpha2"
+	"github.com/vmware-tanzu/antrea/pkg/controller/crdmirroring/types"
+	legacycore "github.com/vmware-tanzu/antrea/pkg/legacyapis/core/v1alpha2"
+	legacycrdclientset "github.com/vmware-tanzu/antrea/pkg/legacyclient/clientset/versioned"
+	legacycorelister "github.com/vmware-tanzu/antrea/pkg/legacyclient/listers/core/v1alpha2"
+)
+
```

Comment:
@hongliangl @tnqn too late to change anything now, but isn't there a dynamic client for K8s that would have enabled us to write the CRD mirroring code with less CRD-specific code (i.e. without relying on the typed clients)?